src/plugins/intel_gpu/src/graph/impls/ocl/non_max_suppression.cpp

```diff
@@ -43,7 +43,9 @@ struct non_max_suppression_impl : typed_primitive_impl_ocl<non_max_suppression>
             args.inputs.push_back(instance.soft_nms_sigma_mem());
         }
 
-        args.outputs = {instance.output_memory_ptr()};
+        for (size_t i = 0; i < instance.outputs_memory_count(); i++) {
```

Comment:
Could you please add comments of this and below, for making future refactoring easier?

Reply:
done